.github/workflows/ci.yml

```diff
@@ -52,75 +63,80 @@ jobs:
       - name: Store Test Results
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: test-results
+          name: backend-test-results
           path: /tmp/test-results
       - name: Store Coverage Results
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: coverage
+          name: backend-coverage
           path: coverage.xml
 
   frontend-lint:
     runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 1
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
         with:
           node-version: ${{ env.NODE_VERSION }}
           cache: 'yarn'
       - name: Install Dependencies
         run: |
           npm install --global --force yarn@1.22.19
-          yarn cache clean && yarn --frozen-lockfile --network-concurrency 1
+          yarn cache clean
+          yarn --frozen-lockfile --network-concurrency 1
       - name: Run Lint
         run: yarn lint:ci
       - name: Store Test Results
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: test-results
+          name: frontend-test-results
           path: /tmp/test-results
 
   frontend-unit-tests:
     runs-on: ubuntu-22.04
     needs: frontend-lint
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 1
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
         with:
           node-version: ${{ env.NODE_VERSION }}
           cache: 'yarn'
       - name: Install Dependencies
         run: |
           npm install --global --force yarn@1.22.19
-          yarn cache clean && yarn --frozen-lockfile --network-concurrency 1
+          yarn cache clean
+          yarn --frozen-lockfile --network-concurrency 1
       - name: Run App Tests
         run: yarn test
       - name: Run Visualizations Tests
-        run: cd viz-lib && yarn test
+        run: |
+          cd viz-lib
+          yarn test
       - run: yarn lint
 
   frontend-e2e-tests:
     runs-on: ubuntu-22.04
     needs: frontend-lint
     env:
-      COMPOSE_FILE: .ci/docker-compose.cypress.yml
-      COMPOSE_PROJECT_NAME: cypress
-      PERCY_TOKEN_ENCODED: ZGRiY2ZmZDQ0OTdjMzM5ZWE0ZGQzNTZiOWNkMDRjOTk4Zjg0ZjMxMWRmMDZiM2RjOTYxNDZhOGExMjI4ZDE3MA==
-      CYPRESS_PROJECT_ID_ENCODED: OTI0Y2th
-      CYPRESS_RECORD_KEY_ENCODED: YzA1OTIxMTUtYTA1Yy00NzQ2LWEyMDMtZmZjMDgwZGI2ODgx
       CYPRESS_INSTALL_BINARY: 0
       PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: 1
+      INSTALL_GROUPS: main
+      COMPOSE_PROFILES: e2e
+      REDASH_PRODUCTION: "true"
+      PERCY_TOKEN: ddbcffd4497c339ea4dd356b9cd04c998f84f311df06b3dc96146a8a1228d170
+      CYPRESS_PROJECT_ID: 924cka
+      CYPRESS_RECORD_KEY: c0592115-a05c-4746-a203-ffc080db6881
```

Comment:
Same question as above. :wink:

Reply:
these values were just b64 encoded. need to move them to secrets

Reply:
I can remove these values if you can move them to secrets

Reply:
This sounded familiar, so after some looking I've found previous discussion about those values:

https://github.com/getredash/redash/commit/f3892e00a5cfb91e1b5384f647101894a6ecef0f#commitcomment-120444257

Looks like they were put there on purpose when we were previously using CircleCI.

```
... this was intentional for CircleCI, because secrets don't work with PRs from Forked repositories.
```

We've moved from CircleCI to GitHub Actions, but the same thing might apply.  We'd probably want to double check though, just to be sure.

Reply:
github actions with `pull_request` event type have the same behavior as in CircleCI, but we can try `pull_request_target`, which requires more attention to PR before running it, cause it provides a token with a write access to base repo, but at provides an access to base repo secrets